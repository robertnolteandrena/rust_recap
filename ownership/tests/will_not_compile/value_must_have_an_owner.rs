fn main() {
    let b;
    {
        let a = String::from("Hello World!");
        b = &a;
        //a goes out of scope here
        //the value no longer has an owner
    }
    println!("b = {b}"); //The value is invalid
}
