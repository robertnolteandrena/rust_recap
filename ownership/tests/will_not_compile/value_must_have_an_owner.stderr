error[E0597]: `a` does not live long enough
 --> tests/will_not_compile/value_must_have_an_owner.rs:5:13
  |
4 |         let a = String::from("Hello World!");
  |             - binding `a` declared here
5 |         b = &a;
  |             ^^ borrowed value does not live long enough
...
8 |     }
  |     - `a` dropped here while still borrowed
9 |     println!("b = {b}"); //The value is invalid
  |                   --- borrow later used here
